### ISSUE TITLE
[Android] alreadyRequestedAccessToContacts

### ISSUE THREAD

**Issue by paintedbicycle** [joshuapinter/react-native-unified-contacts#73]
Hey,

So I've been trying to get the alreadyRequestedAccessToContacts method working on Android and I'm starting to wonder if it's actually possible. I've read a lot and I've tried to dive into the code from other react native libraries that do similar things. Many of them have functions that appear to indeed return an undefined state (not just granted or denied) which could be used in our case. But when I start to audit their code, I see where it's coming from in iOS but in Android there doesn't seem to be that third case.

In my app, I have three states:
- An "Allow access to your contacts" button (for before they've interacted with permissions)
- The list of contacts (if permission granted)
- A "You have chosen to not allow contacts. If you change your mind, click here" button that takes you to the permissions page if they rejected permission
I've read stuff like this https://inthecheesefactory.com/blog/things-you-need-to-know-about-android-m-permission-developer-edition/en that nicely walks through the whole process, but again skips the already requested mode. So do the official docs https://developer.android.com/training/permissions/requesting#java

I don't think it's possible to create those 3 different states without this "undefined" state for permissions. I could probably save something to the db myself, but this seems like something a contacts library should handle, no?

Now - React Native Permissions https://github.com/yonahforst/react-native-permissions again lists 4 different permissions states (the 3 above plus restricted) and they seem to have a decent pattern: https://github.com/yonahforst/react-native-permissions/blob/master/lib/permissions.android.js#L56

Is that something that could work here?

**Comment by paintedbicycle** [comment]
Very interesting. I’ll test this tomorrow. 

I’m down for whatever solution makes sense. Even a permissionsRequestState that can return the 4 states would be good. Just need to be able to recreate all 3 scenarios in my app, including never asked. 

Paul

**Comment by paintedbicycle** [comment]
Is there only a promise version? Non promise version doesn’t appear public and there is no promise iOS version yet so it’s a mismatch. 

Paul

**Comment by joshuapinter** [comment]
Non-Promise version of `alreadyRequestedAccessToContacts` added!

**Comment by paintedbicycle** [comment]
Just tested. Works well! Confirmed fixed.
